api/src/data_migration/transformation/transform_oracle_data.py

```diff
@@ -0,0 +1,180 @@
+import logging
+from enum import StrEnum
+from typing import Tuple, Type, TypeVar, cast
+
+from sqlalchemy import select
+
+from src.adapters.db import PostgresDBClient
+from src.db.models.base import Base
+from src.db.models.opportunity_models import Opportunity
+from src.db.models.transfer.topportunity_models import TransferTopportunity
+from src.task.task import Task
+
+S = TypeVar("S", bound=Base)
+D = TypeVar("D", bound=Base)
+
+from src.constants.lookup_constants import OpportunityCategory
+
+logger = logging.getLogger(__name__)
+
+
+class TransformOracleData(Task):
```

Comment:
@jamesbursa - FYI I'm imagine that the script we'll have for the copy + transform will just look like:

```
def whatever_entrypoint_func(db_session):
     ExtractLoadOracleData(db_session).run()
     TransformOracleData(db_session).run()
```